privacy_estimates/experiments/components/postprocess_dpd_data/postprocess_dpd_data.py

```diff
@@ -0,0 +1,56 @@
+import logging
+import numpy as np
+from datasets import Dataset, features
+from mldesigner import command_component, Input, Output
+from pathlib import Path
+from json import load, dump
+
+
+logger = logging.getLogger(__name__)
+
+
+@command_component(
+    display_name="Postprocess DP Distinguisher data",
+    environment="environment.aml.yaml",
+)
+def postprocess_dpd_data(dpd_data: Input, dp_parameters: Input, seed: int, scores: Output, challenge_bits: Output,
+                         postprocessed_dp_parameters: Output(type="uri_file")):  # noqa: F821
+    """
+    Post-processing for differential privacy distinguisher
+
+    We follow Algorithm 2 in https://arxiv.org/abs/2302.07956. We rearrange
+    <g'|\nabla'[t] + g'> to <g'|\nabla'[t]> + <g'|g'> which allows us to pull
+    <g'|g'> = C^2 (which is the gradient clipping norm) out of the training algorithm.
+    Instead we add the C^2 term in this script.
+    """
+    with Path(dpd_data).open() as f:
+        raw_data = load(f)
+    sensitivity = raw_data["sensitivity"]
+
+    with Path(dp_parameters).open() as f:
+        dp_parameters = load(f)
+
+    # Differential Privacy Distinguisher 
```

Comment:
Do these need to be manually set with varying adversary models?

Reply:
This component is called by the DPD game. We're doing this since the canary gradient attack from https://arxiv.org/pdf/2302.07956.pdf doesn't test the subsampling since the canary is inserted independently from the batch creation. Since it's independent we can do it in post. We can set the sampling probability to 1 to get the most samples from the training run but we could also set it to another value later if we wanted to.